stl/inc/functional

```diff
@@ -1479,6 +1479,69 @@ namespace placeholders {
     _INLINE_VAR constexpr _Ph<20> _20{};
 } // namespace placeholders
 
+#if _HAS_CXX20
+// FUNCTION TEMPLATE _Call_front_binder
+template <size_t... _Ix, class _Cv_FD, class _Cv_tuple_TiD, class... _Unbound>
+auto _Call_front_binder(index_sequence<_Ix...>, _Cv_FD&& _Obj, _Cv_tuple_TiD&& _Tpl, _Unbound&&... _Unbargs) {

```

Comment:
I believe that the occurrences of `auto` as a return type are incorrect and that they should be `decltype(auto)`. A function returning `Meow&` should have that preserved, instead of decayed to `Meow`. (This applies only to `_Call_front_binder` and `operator()`. `bind_front` itself is correct to use `auto`.)

Reply:
Thanks, it is correct to use `decltype(auto)`!
Also, after messing around with some `static_asserts()`, I discovered this failed.
```    
    auto add = [](int a, int b) {return a + b;};
    auto bound = std::bind_front(add, 1);
    static_assert(!std::is_invocable_t<decltype(bound), int, int>); // this should not give a compiler error!
```
So, would it be best to specify the return type within the `decltype()` to avoid the compiler error? e.g.
```
    template <class... _Unbound>
    constexpr auto operator()(_Unbound&&... _Unbargs)
    & noexcept(is_nothrow_invocable_v<decay_t<_Fx>&, decay_t<_Types>&..., _Unbound...>) 
    -> decltype(_Call_front_binder(_Seq{}, _Mypair._Get_first(), _Mypair._Myval2, _STD forward<_Unbound>(_Unbargs)...)) {
        return _Call_front_binder(_Seq{}, _Mypair._Get_first(), _Mypair._Myval2, _STD forward<_Unbound>(_Unbargs)...);
    }
```